pkg/ccl/crosscluster/physical/stream_ingestion_dist.go

```diff
@@ -588,23 +588,49 @@ func (p *replicationFlowPlanner) constructPlanGenerator(
 	}
 }
 
-func getNodes(plan *sql.PhysicalPlan) (src, dst map[string]struct{}, nodeCount int) {
-	dst = make(map[string]struct{})
-	src = make(map[string]struct{})
-	count := 0
-	for _, proc := range plan.Processors {
-		if proc.Spec.Core.StreamIngestionData == nil {
```

Comment:
this bug affects both PCR and LDR and was introduced once PCR start using multiple Procs per node. The core problem is this count var is incorrect-- `getNodes` needs to return the count of nodes, not the count of procs. Could we simply refactor this via the following instead ? 

```
diff --git a/pkg/ccl/crosscluster/logical/logical_replication_job.go b/pkg/ccl/crosscluster/logical/logical_replication_job.go
index 122508d536d..1b6753c8494 100644
--- a/pkg/ccl/crosscluster/logical/logical_replication_job.go
+++ b/pkg/ccl/crosscluster/logical/logical_replication_job.go
@@ -249,10 +249,15 @@ func getNodes(plan *sql.PhysicalPlan) (src, dst map[string]struct{}, nodeCount i
                        // Skip other processors in the plan (like the Frontier processor).
                        continue
                }
-               dst[proc.SQLInstanceID.String()] = struct{}{}
-               count += 1
-               src[proc.Spec.Core.LogicalReplicationWriter.PartitionSpec.PartitionID] = struct{}{}
-               count += 1
+               if _, ok := dst[proc.SQLInstanceID.String()]; !ok {
+                       dst[proc.SQLInstanceID.String()] = struct{}{}
+                       count += 1
+               }
+               if _, ok := src[proc.Spec.Core.LogicalReplicationWriter.PartitionSpec.PartitionID]; !ok {
+                       src[proc.Spec.Core.LogicalReplicationWriter.PartitionSpec.PartitionID] = struct{}{}
+                       count += 1
+               }
+
        }
        return src, dst, count
 }

```